src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -79,24 +79,13 @@ public static void SetConfirmationId(this IBasicProperties properties, ulong con
         public static bool TryGetConfirmationId(this IBasicProperties properties, out ulong confirmationId)
         {
             confirmationId = 0;
-
-            if (properties.Headers.ContainsKey(confirmationIdHeader))
-            {
-                var headerBytes = properties.Headers[confirmationIdHeader] as byte[];
-                var headerString = Encoding.UTF8.GetString(headerBytes ?? new byte[0]);
-
-                return UInt64.TryParse(headerString, out confirmationId);
-            }
-
-            return false;
+            object value;
+            return properties.Headers.TryGetValue(confirmationIdHeader, out value) &&
```

Comment:
Also, I know this has come up before too, but I'd personally prefer to see this on one line instead of wrapped to two, but that's just my preference.